comment/mixins.py

```diff
@@ -49,7 +51,17 @@ def has_permission(self, request):
         return super().has_permission(request)
 
 
-class BaseCommentMixin(BaseCommentPermission):
+class GetObjectMixin:
+    def get_object(self):
+        try:
+            self.comment = Comment.objects.select_related('user', 'reaction', 'flag').get(pk=self.kwargs.get('pk'))
+        except Comment.DoesNotExist:
+            raise Http404
```

Comment:
this is how it is being handled prior to this patch, in case this needs to be done, it should ideally be done in a separate ticket. Also, as you may see this `mixin` is also being used by `rest_framework` . hence, it will need some sort of argument like `api` to differentiate and render an appropriate response.

Reply:
Agree. I believe this was missed in the last update when we changed the response to json. We can create a ticket for this and IMO it should be done ASAP to ensure responses consistency.

Reply:
Also, for what is worth, this boolean argument `api` should be a keyword-only types to ensure we do not get into [`boolean trap`](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap). I have already created a couple of them in our codebase, we can probably get rid of them in `v3.0.0`.

Reply:
Do you want to fix it in this pull-request and add this as a separate commit? As it is, I will have to rebase this(which is kind of a bit annoying to be honest :( )

Reply:
Either way fine by me.

Reply:
Hey @Radi85, we can't directly return `UTF8JsonResponse` because the return value is getting consumed by the `get` and `post` methods in the `view` as `self.comment`. As of now, I can't think of an elegant way to fix this issue for non-DRF version.

One of the ways is to raise `CommentBadRequest` exception and catch it inside the `view` but since this method is being consumed by a lot of `views`'s methods it would mean duplication of a lot of code.

I can't seem to think of a way to handle this without raising an exception, but i'm not sure what and how to do it at the moment. Any ideas from your side are welcome.

Reply:
one of the probable ways to handle this for the `view` is to create a `dispatch` method on the newly created `GetObjectMixin` class. We now need to call `get_object` from the `dispatch` method(to detect if the object is present and return a 404 JSON response). This opens up another can of worms, especially for rest framework because we also need to call the method `get_object_permissions` manually from `get_object` since we have already overridden it. You may refer to the [documentation regarding it](https://www.django-rest-framework.org/api-guide/permissions/#object-level-permissions). 

Now when we are calling the `get_object_permissions` method from `get_object`, we also need to pass in the `request` object which although is available as a part of the currently calling object(`self`), but not all attributes that are required are set on it. [The `dispatch` method of the `APIView` class for rest framework also sets from attributes to the `request` object internally](https://github.com/encode/django-rest-framework/blob/master/rest_framework/views.py#L492-L497), which are then required by other methods such as `get_object_permissions`.

Due to the complexity of the task involved, I'm inclined to say a `wontfix` here and probably transfer this issue to a discussion so that we can probably refer to this in the future, in case it is required.

Reply:
I would suggest to keep the old implementation for getting comment object and concentrate on reducing the number of queries in this PR.

Reply:
As of now, this patch is intended to reduce the number of queries. I'm not sure how we can keep the old implementation without creating a `mixin` or something similar for that matter.  From what I can comprehend, I see keeping the old method would require replicating the code multiple times, since the `comment` object is required by a lot of `views`.

Reply:
Im not sure why we would need a mixin to reduce the number of queries?! I understood that the mixin you used is to reduce the duplicated code which we can do that in a separate PR if we still see it necessary.
Also, we already have ObjectLevelMixin class for implementing get_object; now using a different mixin for the same purpose is not ideal.
As I said, if the refactoring is needed, we can look into this class first and we might subclass it for our purpose.

Reply:
okay. seems legit. had we arrived at this conclusion a bit earlier, it could have probably saved a bit to and forth cycles. 😄